pkg/volume/csi/expander_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"os"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+func TestNodeExpand(t *testing.T) {
+	tests := []struct {
+		name          string
+		nodeExpansion bool
+		nodeStageSet  bool
+		volumePhase   string
+		success       bool
+	}{
+		{
+			name:    "when node expansion is not set",
+			success: false,
+		},
+		{
+			name:          "when nodeExpansion=on, nodeStage=on, volumePhase=staged",
```

Comment:
also tests for volumePhase=published

Reply:
I will add for completeness but we aren't really checking for published phase so we should be good anyways.